test/json/test-functions.json

```diff
@@ -0,0 +1,259 @@
+{
+  "integer": [
+    { "src": "hello {4.2 :integer}", "exp": "hello 4" },
+    { "src": "hello {-4.20 :integer}", "exp": "hello -4" },
+    { "src": "hello {0.42e+1 :integer}", "exp": "hello 4" },
+    {
+      "src": ".match {$foo :integer} one {{one}} * {{other}}",
+      "params": { "foo": 1.2 },
+      "exp": "one"
+    },
+    {
+      "src": ".match {$foo :integer} 1.2 {{=1.2}} one {{one}} * {{other}}",
+      "params": { "foo": 1.2 },
+      "exp": "=1.2"
+    },
+    {
+      "src": ".match {$foo :integer} 1.2 {{=1.2}} |1,2| {{=1,2}} * {{other}}",
+      "locale": "fr",
+      "params": { "foo": 1.2 },
+      "exp": "=1.2"
+    }
+  ],
+  "number": [
```

Comment:
I would move `:number` ahead of `:integer`. Note that you don't test any of `:integer`'s options, e.g. `minimumIntegerDigits`, `signDisplay`, et al.